lib/excon/socket.rb

```diff
@@ -61,9 +63,22 @@ def read(max_length = nil)
 
     def readline
       if @nonblock && RUBY_VERSION.to_f > 1.8_7
-        buffer = String.new
-        buffer << (read_nonblock(1) || raise(EOFError)) while buffer[-1] != "\n"
-        buffer
+        result = String.new
+        block = @read_buffer
+        @read_buffer = String.new
```

Comment:
I believe `@read_buffer` should already be setup from `initialize`. Is there a particular reason to overwrite that here?

Reply:
Got it. Yeah, I think I like the inline version you suggest.